webapp/channels/src/plugins/registry.ts

```diff
@@ -125,6 +125,25 @@ export default class PluginRegistry {
         return dispatchPluginComponentAction('PostMessageAttachment', this.id, component);
     });
 
+    // Register components for search.
+    // Accepts React components. Returns a unique identifier.
+    registerSearchComponents = reArg(['buttonComponent', 'suggestionsComponent', 'hintsComponent', 'action'], ({buttonComponent, suggestionsComponent, hintsComponent, action}: any) => {
```

Comment:
0/5 but perhaps we should just have this method take a map instead of using `reArg` to prevent people from calling the new API in the old way

Reply:
I was replicating the existing API, but yes, I can change it.

Reply:
I just removed the `reArg` call, so now we should just have a function that receives an object.